### PR TITLE
Zone Manager: "Show All / Hide Distant" highlights selection

### PR DESCRIPTION
**Pull request by ShnitzelX2**
#### Summary
Interface "Zone Manager: "Show All / Hide Distant" highlights selection"

#### Purpose of change

Followup for #74435

It was difficult to tell whether zones were hidden or not in the zone manager. Now, the appropriate word (Show **All** / Hide **Distant**) is highlighted to indicate the current selection.

#### Describe the solution

See above.

#### Describe alternatives you've considered

Originally, the number of zones along with the closest hidden zone would display if applicable. However, with @PatrikLundell's input and after playtesting for a while, I came to the conclusion that neither feature was useful often enough to merit taking up a line of UI space.

#### Testing

Basically played a game with the new feature. It helps.

Hide Distant
image
Show All
image

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -6920,8 +6921,22 @@ static void zones_manager_shortcuts( const catacurses::window &w_info, faction_i
     shortcut_print( w_info, point( tmpx, 4 ), c_white, c_light_green, _( "<Enter>-Edit" ) );
 
     tmpx = 1;
+
+    std::string all_zones = "all";
```
Comment (Qrox) [inline src/game.cpp:6925]: Please do no split `Show all` etc into two words because other languages do not necessarily put words in that order or use exactly two words.

Comment (ShnitzelX2) [reply src/game.cpp]: @Qrox should be good to go, I have left the strings intact

File: src/game.cpp [hunk 2]
```diff
@@ -6724,8 +6725,20 @@ static void zones_manager_shortcuts( const catacurses::window &w_info, faction_i
     shortcut_print( w_info, point( tmpx, 4 ), c_white, c_light_green, _( "<Enter>-Edit" ) );
 
     tmpx = 1;
-    tmpx += shortcut_print( w_info, point( tmpx, 5 ), c_white, c_light_green,
-                            _( "<S>how all / hide distant" ) ) + 2;
+    std::string all_zones = "<S>how all";
+    std::string distant_zones = "Hide distant";
```
Comment (Qrox) [inline src/game.cpp:6729]:
```suggestion
    std::string all_zones = _( "<S>how all" );
    std::string distant_zones = _( "Hide distant" );
```
Marks the strings for translation.

File: src/game.cpp [hunk 3]
```diff
@@ -6724,8 +6725,20 @@ static void zones_manager_shortcuts( const catacurses::window &w_info, faction_i
...
+    selection_color[int( !show_all_zones )] = c_yellow;
+
+    nc_color current_color = selection_color[0];
+    tmpx += shortcut_print( w_info, point( tmpx, 5 ), current_color, c_light_green, all_zones );
+    current_color = c_white;
+    print_colored_text( w_info, point( tmpx, 5 ), current_color, current_color, " / " );
+    tmpx += 3;
+    current_color = selection_color[1];
+    print_colored_text( w_info, point( tmpx, 5 ), current_color, current_color, distant_zones );
+    tmpx += distant_zones.length() + 2;
```
Comment (Qrox) [inline src/game.cpp:6740]:
```suggestion
    tmpx += utf8_width( distant_zones ) + 2;
```
`length()` is not equal to display width for multi-byte characters.
